CONTRIBUTING.md

```diff
@@ -228,28 +217,31 @@ smoothly, but it allows you to help out by noticing when a step is
 missed or to learn in case someday you become a project maintainer as
 well!
 
-### Architecture
-
-At a high level we have a bunch of folders. Each high level is described below;
-
-- src\extension.node.ts = entry point for node extension
-- src\extension.web.ts = entry point for web extension
-- src\kernels = code related to executing jupyter kernels
-- src\notebooks = code related to vscode notebook UI
-- src\interactive-window = interactive window
-- src\platform = grab bag of utilities common to rest of code
-- src\web-views = code related to web views we create. Variable view, data viewer, and plot viewer are all here. Both extension and UI side code.
-- src\intellisense = code related to starting pylance and combining pylance completions with kernel completions
-- src\test = all test related code
-- src\telemetry = all code related to gathering and sending telemetry
-- build = build scripts and config files
-- .github = github workflows and issue templates
-- news = markdown files used to generate the next release's changelog
-- pythonFiles = python files used to do things like view data in the data viewer
-- typings = .d.ts files used to provide types for node_modules without any typings
-- types = .ts files used to provide types for node_modules
-- images = gifs and jpgs for markdown files
-- docs = generated dependency graph files (obsolete at the moment)
+### Folder Structure
+
+At a high level we have a bunch of folders. Each high level is described in this wiki [page](https://github.com/microsoft/vscode-jupyter/wiki/Source-Code-Organization)
+
+### Typical workflow
+
+Here's an example of a typical workflow:
+
+1. Sync to main (get your fork's main to match vscode-jupyter's main)
+1. Create branch
+1. `npm ci`
+1. `npm run clean`
+1. Start VS code Insiders root
+1. CTRL+SHIFT+B and build `Compile Web Views` and `Compile`
+1. Make code changes
+1. Write and [run](https://github.com/microsoft/vscode-jupyter/blob/29c4be79f64df1858692321b43c3079bb77bdd69/.vscode/launch.json#L252) unit tests if appropriate
+1. Test with [`Extension`](https://github.com/microsoft/vscode-jupyter/blob/29c4be79f64df1858692321b43c3079bb77bdd69/.vscode/launch.json#L6) launch task
+1. Repeat until works in normal extension
+1. Test with [`Extension (web)`](https://github.com/microsoft/vscode-jupyter/blob/29c4be79f64df1858692321b43c3079bb77bdd69/.vscode/launch.json#L34) launch task
+1. Run [jupyter notebook server](https://github.com/microsoft/vscode-jupyter/wiki/Connecting-to-a-remote-Jupyter-server-from-vscode.dev) to use in web testing
+1. Repeat until works in web extension
+1. Write integration tests and [run](https://github.com/microsoft/vscode-jupyter/blob/29c4be79f64df1858692321b43c3079bb77bdd69/.vscode/launch.json#L216) locally.
+1. Submit PR
+1. Check PR output to make sure tests don't fail.
+1. Debug [CI test failures](https://github.com/microsoft/vscode-jupyter/wiki/Tests)
```

Comment:
Going to work on this wiki page next. It's super out of date. I'll change it to explain how to debug tests locally and on CI